src/App/Fossa/Container.hs

```diff
@@ -0,0 +1,198 @@
+      <*> obj .: "version"
+      <*> obj .: "type"
+      <*> obj .: "purl"
+      <*> obj .: "metadataType"
+      -- We delete "files" as early as possible, which reduces
+      -- the size by over 95% in many cases.
+      -- We use Lazy delete to avoid evaluating the innards of
+      -- the field, since Aeson will try to avoid evaluating it
+      -- as well.
+      <*> (LMap.delete "files" <$> obj .: "metadata")
+
+newtype ResponseSource
+  = ResponseSource
+      {sourceTarget :: SourceTarget}
+
+instance FromJSON ResponseSource where
+  parseJSON = withObject "ResponseSource" $ \obj ->
+    ResponseSource <$> obj .: "target"
+
+data ResponseDistro
+  = ResponseDistro
+      { distroName :: Text,
+        distroVersion :: Text
+      }
+
+instance FromJSON ResponseDistro where
+  parseJSON = withObject "ResponseDistro" $ \obj ->
+    ResponseDistro <$> obj .: "name"
+      <*> obj .: "version"
+
+data SourceTarget
+  = SourceTarget
+      { targetDigest :: Text,
+        targetTags :: [Text]
+      }
+
+instance FromJSON SourceTarget where
+  parseJSON = withObject "SourceTarget" $ \obj ->
+    SourceTarget <$> obj .: "digest"
+      <*> obj .: "tags"
+
+-- | The reorganized output of syft into a slightly different format
+data ContainerScan
+  = ContainerScan
+      { imageData :: ContainerImage,
+        imageTag :: Text,
+        imageDigest :: Text
+      }
+
+instance ToJSON ContainerScan where
+  toJSON scan = object ["image" .= imageData scan]
+
+data ContainerImage
+  = ContainerImage
+      { imageArtifacts :: [ContainerArtifact],
+        imageOs :: Text,
+        imageOsRelease :: Text
+      }
+
+instance ToJSON ContainerImage where
+  toJSON ContainerImage {..} =
+    object
+      [ "os" .= imageOs,
+        "osRelease" .= imageOsRelease,
+        "artifacts" .= imageArtifacts
+      ]
+
+data ContainerArtifact
+  = ContainerArtifact
+      { conArtifactName :: Text,
+        conArtifactVersion :: Text,
+        conArtifactType :: Text,
+        conArtifactPkgUrl :: Text,
+        conArtifactMetadataType :: Text,
+        conArtifactMetadata :: Map Text Value
+      }
+
+instance ToJSON ContainerArtifact where
+  toJSON ContainerArtifact {..} =
+    object
+      [ "name" .= conArtifactName,
+        "fullVersion" .= conArtifactVersion,
+        "type" .= conArtifactType,
+        "purl" .= conArtifactPkgUrl,
+        "metadataType" .= conArtifactMetadataType,
+        "metadata" .= LMap.delete "files" conArtifactMetadata
+      ]
+
+extractRevision :: OverrideProject -> ContainerScan -> ProjectRevision
+extractRevision OverrideProject {..} ContainerScan {..} = ProjectRevision name revision Nothing
+  where
+    name = fromMaybe imageTag overrideName
+    revision = fromMaybe imageDigest overrideRevision
+
+
+toContainerScan :: Has Diagnostics sig m => SyftResponse -> m ContainerScan
+toContainerScan SyftResponse {..} = do
+  let convertArtifact (ResponseArtifact a b c d e f) = ContainerArtifact a b c d e f
```

Comment:
[no action] I hate it but I like that we're keeping the internal type separate from the transport type

Reply:
I hated this too, and so I made it more explicit.